iOS/DuckDuckGo/DuckPlayer/DuckPlayerSettings.swift

```diff
@@ -235,46 +275,98 @@ final class DuckPlayerSettingsDefault: DuckPlayerSettings {
 
     /// The current mode of Duck Player.
     var mode: DuckPlayerMode {
-        if isFeatureEnabled {
+        get {
+            guard isFeatureEnabled else { return .disabled }
+            // Return the underlying setting, reflecting user choice or variant default
             return appSettings.duckPlayerMode
-        } else {
-            return .disabled
+        }
+        set {
+            // Allow direct setting if needed, but primarily managed by variant
+            if newValue != appSettings.duckPlayerMode {
+                appSettings.duckPlayerMode = newValue
+                triggerNotification()
+            }
         }
     }
 
     /// Indicates if the "Always Ask" overlay has been hidden.
     var askModeOverlayHidden: Bool {
-        if isFeatureEnabled {
-            return appSettings.duckPlayerAskModeOverlayHidden
-        } else {
-            return false
+        get {
+            if isFeatureEnabled {
+                return appSettings.duckPlayerAskModeOverlayHidden
+            } else {
+                return false
+            }
+        }
+        set {
+            if newValue != appSettings.duckPlayerAskModeOverlayHidden {
+                appSettings.duckPlayerAskModeOverlayHidden = newValue
+                triggerNotification()
+            }
         }
     }
+    
 
     /// Flag to allow the first video to play without redirection.
     var allowFirstVideo: Bool = false
 
     /// Determines if Duck Player should open videos in a new tab.
     var openInNewTab: Bool {
-        return appSettings.duckPlayerOpenInNewTab
+        get {
+            // Return the underlying AppSetting value directly
+            return appSettings.duckPlayerOpenInNewTab
+        }
+        set {
+            // Allow direct setting if needed, potentially overridden by variant change
+            if newValue != appSettings.duckPlayerOpenInNewTab {
+                appSettings.duckPlayerOpenInNewTab = newValue
+                triggerNotification()
+            }
+        }
     }
 
     // Determines if we should use the native verion of DuckPlayer (Internal only)
     var nativeUI: Bool {
-        return appSettings.duckPlayerNativeUI && internalUserDecider.isInternalUser && UIDevice.current.userInterfaceIdiom == .phone
+        get {
+            guard internalUserDecider.isInternalUser, UIDevice.current.userInterfaceIdiom == .phone else { return false }
```

Comment:
Maybe extracting this check to another var for code reuse and ease of change later?

Reply:
This will also go away as soon as we release the experiment